JavaScript/JavaScriptSDK/Util.ts

```diff
@@ -580,19 +580,24 @@ module Microsoft.ApplicationInsights {
         /**
         * Checks if a request url is not on a excluded domain list and if it is safe to add correlation headers
         */
-        public static canIncludeCorrelationHeader(config: IConfig, requestUrl: string) {
+        public static canIncludeCorrelationHeader(config: IConfig, requestUrl: string, currentHost: string) {
             if (config && config.disableCorrelationHeaders) {
                 return false;
             }
 
+            let requestHost = UrlHelper.parseUrl(requestUrl).host;
+            if (config && !config.enableCorsCorrelation && requestHost !== currentHost) {
```

Comment:
probably should ignore case sensitivity.

Reply:
yes. thanks!